src/tyr/matrix_serializer.cc

```diff
@@ -118,6 +145,17 @@ json::ArrayPtr serialize_row(const valhalla::Matrix& matrix,
       if (!date_time.empty()) {
         map->emplace("date_time", date_time);
       }
+      // TODO(nils): shapes aren't implemented yet in TDMatrix
+      if (matrix.shapes().size() && !matrix.shapes()[i].empty()) {
+        switch (shape_format) {
+          case geojson:
+            map->emplace("shape",
```

Comment:
ok so for this serializer every entry now gets an optional shape output. i guess for failed pairs even if shape was requested we dont output anything?

Reply:
i guess my point is that disagrees with the comment in the valhalla serializer about always having an entry null or not

Reply:
yes, for the verbose version (default), we don't even put a shape, also not for failed pairs. for the terser verbose=false, we do have to keep the `nullptr` in the response so it's properly parsable.